ECS test suite

willthames opened the pull request:
##### SUMMARY

Tests for:
* ecs_cluster
* ecs_service
* ecs_service_facts
* ecs_taskdefinition
* ecs_taskdefinition_facts

Initial test suite to allow work on ECS modules to be performed in a safer fashion

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME
ECS integration tests 

##### ANSIBLE VERSION

```
ansible 2.5.0 (devel 51475cd623) last updated 2017/12/11 13:41:43 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.14 (default, Nov  2 2017, 18:42:05) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

##### ADDITIONAL INFORMATION

ecsServiceRole and ecsInstanceRole will need to be created manually for the CI (or anyone using testing-policies) - the role runs iam_role against them to check that they meet the expected permissions, not necessarily to create them (although with the right permissions, they would be created)

Commit:
ECS test suite

Tests for:
* ecs_cluster
* ecs_service
* ecs_service_facts
* ecs_taskdefinition
* ecs_taskdefinition_facts

gundalow commented on test/integration/targets/ecs_cluster/meta/main.yml:
```diff
@@ -0,0 +1,3 @@
+dependencies:
+  - prepare_tests
```
I think prepare_tests is empty

willthames replied:
Lazy copy and paste on my part. setup_ec2 isn't needed either unless you then include setup_ec2/tasks/common.yml, which most targets don't do.

gundalow commented on test/integration/targets/ecs_cluster/tasks/main.yml:
```diff
@@ -0,0 +1,319 @@
+        - AmazonEC2ContainerServiceRole
+        <<: *aws_connection_info
+
+    - name: create an ECS cluster
```
Can we do another create after this and ensure no change (idempotent)?

willthames replied:
Sure, I'll have to add FIXMEs for the things that do change (ecs_cluster is fine, ecs_service and ecs_taskdefinition are broken in this respect)

willthames committed:
Add idempotency testing

Test ecs_cluster, ecs_service and ecs_taskdefinition for trivial
idempotency. Add FIXMEs to the tests because the latter two fail.

Remove unused dependencies

willthames commented:
@gundalow ready_for_review
